### DISCUSSION TITLE
Multiple Schedulers

### DISCUSSION THREAD

**Discussion by udayaDJ** [agronholm/apscheduler#765] [Q&A]
I am trying to create an Job scheduler feature and came across APScheduler. I have few doubts that I've listed down which I couldn't gather much infos.

- I heard that with multiple schedulers running(Assume 3 schedulers are running), if we add a job(Job A) using a particular scheduler(scheduler 1) this particular scheduler is responsible for the job execution. Scheduler 2 won't be taking Job A. Is that true?

- Incase if I am using multiple scheduler with a single job store say mongodb, will there be any duplication in the job execution(I would be running this in a fastapi server)?
 
- Do the APScheduler have failover mechanism incase a scheduler went down and can we make the other schedulers to take the jobs of scheduler that went down.

And I am thinking of using Celery for the job execution. Basically the job trigger a function which in-turn calls the celery worker. Is this an effective way to execute the jobs? Will there be any limitation/blocking with APScheduler?

Tried going through multiple blogs and docs. Have tried the celery part, just wanted to know if the way i'm going is correct.

**Comment by kyl33r** [comment]
hey are there any updates on this

**Comment by legout** [comment]
When you are using apscheduler v4 with a data store and event broker, you can have several schedulers and workers and a job is only run once, however you are not able to control which worker is responsible for the job execution.
